Sources/SwiftParserDiagnostics/ParserDiagnosticMessages.swift

```diff
@@ -287,6 +287,26 @@ public struct CannotParseVersionTuple: ParserError {
   }
 }
 
+public struct ConnectedExpectedNodeAndUnexpectedNodesError: ParserError {
+  public let expectedNode: Syntax
+  public let unexpectedNodes: UnexpectedNodesSyntax
+
+  public var message: String {
+    if let expectedNodeTypeName = expectedNode.nodeTypeNameForDiagnostics(allowBlockNames: false) {
+      if expectedNode.isMissingAllTokens {
+        return "expected \(expectedNodeTypeName), \(unexpectedNodes.shortSingleLineContentDescription) is not the valid \(expectedNodeTypeName)"
+      } else {
+        if expectedNode.position > unexpectedNodes.position {
+          return "expected \(expectedNodeTypeName), code '\(unexpectedNodes.description + expectedNode.description)' is not the valid \(expectedNodeTypeName)"
+        } else if expectedNode.position < unexpectedNodes.position {
+          return "expected \(expectedNodeTypeName), code '\(expectedNode.description + unexpectedNodes.description)' is not the valid \(expectedNodeTypeName)"
+        }
+      }
+    }
+    return "unexpected \(unexpectedNodes.shortSingleLineContentDescription)"
+  }
+}
```

Comment:
I can't think of a suitable error message and type name 🤔  
If you have any suggestions for a better message, I'll make the changes.